packages/react-ui/components/DateInput/__tests__/DateInput-test.tsx

```diff
@@ -236,4 +242,121 @@ describe('DateInput as InputlikeText', () => {
 
     expect(screen.getByTestId(InputLikeTextDataTids.nativeInput)).toBeDisabled();
   });
+
+  it('should handle keydown event', () => {
+    const onKeyDown = jest.fn();
+    renderRTL(<DateInput onKeyDown={onKeyDown} />);
+
+    fireEvent.keyDown(getInput(), 'a');
+
+    expect(onKeyDown).toHaveBeenCalled();
+  });
+
+  it('should handle onFocus event', () => {
+    const onFocus = jest.fn();
+    renderRTL(<DateInput onFocus={onFocus} />);
+
+    userEvent.click(getInput());
+
+    expect(onFocus).toHaveBeenCalled();
+  });
+
+  it('should handle onBlur event', () => {
+    const onBlur = jest.fn();
+    renderRTL(<DateInput onBlur={onBlur} />);
+    userEvent.tab();
+    expect(screen.getByTestId(InputLikeTextDataTids.root)).toHaveFocus();
+    userEvent.tab();
+    expect(screen.getByTestId(InputLikeTextDataTids.root)).not.toHaveFocus();
+    expect(onBlur).toHaveBeenCalled();
+  });
+
+  it('should handle double click', () => {
+    const inputLikeTextRef = React.createRef<DateInput>();
+
+    renderRTL(<DateInput value="27.04.1988" ref={inputLikeTextRef} />);
+    const input = getInput();
+    userEvent.dblClick(input);
+    expect(screen.getByTestId(InputLikeTextDataTids.root)).toHaveFocus();
+
+    expect(inputLikeTextRef.current?.state.selected).toBe(InternalDateComponentType.All);
+  });
+
+  it('should clear selected text in the input after pressing delete button', () => {
+    renderRTL(<DateInput value="27.04.1988" />);
+    const input = getInput();
+    userEvent.dblClick(input);
+    userEvent.keyboard('{delete}');
+
+    // eslint-disable-next-line jest-dom/prefer-to-have-text-content
+    expect(input.textContent).toBe(
+      `${MASK_CHAR_EXEMPLAR.repeat(2)}.${MASK_CHAR_EXEMPLAR.repeat(2)}.${MASK_CHAR_EXEMPLAR.repeat(4)}`,
+    );
+  });
+
+  it('should fully clear input by backspace pressing', () => {
```

Comment:
Тут ещё один тест с похожим текстом остался

```suggestion
  it('should clear selected text in the input after pressing backspace button', () => {
```